CMakeLists.txt

```diff
@@ -115,6 +97,21 @@ if (APPLE)
     endif()
 endif()
 
+project (ngraph)
+
+# Tells if ngraph source tree is embedded in some other project
+# TRUE if embedded in source tree
+# FALSE if used standalone or as an external project
+set(NGRAPH_IS_EMBEDDED (NOT CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR))
+
+set(NGRAPH_INCLUDE_PATH
+    ${PROJECT_SOURCE_DIR}/src
+)
+
+if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
+    message(FATAL_ERROR "In-source builds are not allowed.")
```

Comment:
did you test that this works as intended?

Reply:
Yes. `cmake .` in the root directory fails as intend.
```
❯ cmake .
-- Found Git: /usr/local/bin/git (found version "2.26.2")
-- Current hash 0ea9a38e7f535dcc843246f491079115861b1a83
-- NGRAPH_VERSION 0.29.0-rc.0+0ea9a38
-- NGRAPH_VERSION_SHORT 0.29.0-rc.0
-- NGRAPH_WHEEL_VERSION 0.29.0rc0
-- NGRAPH_API_VERSION 0.29
-- Detected: OSX 10.15
-- Trying Xcode sdk at /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk.
-- Setting CMAKE_OSX_SYSROOT for macos 10.15 to /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk
-- The C compiler identification is AppleClang 11.0.3.11030032
-- The CXX compiler identification is AppleClang 11.0.3.11030032
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:112 (message):
  In-source builds are not allowed.


-- Configuring incomplete, errors occurred!
```